api/services/observatory/src/providers/DistributionRepositoryProvider.ts

```diff
@@ -0,0 +1,104 @@
+import { provider } from '@ilos/common';
+import { PostgresConnection } from '@ilos/connection-postgres';
+import {
+  DistributionRepositoryInterface,
+  DistributionRepositoryInterfaceResolver,
+  InsertLastMonthDistributionParamsInterface,
+  InsertLastMonthDistributionResultInterface,
+  refreshAllDistributionParamsInterface,
+  refreshAllDistributionResultInterface,
+  JourneysByDistancesParamsInterface,
+  JourneysByDistancesResultInterface,
+  JourneysByHoursParamsInterface,
+  JourneysByHoursResultInterface,
+} from '../interfaces/DistributionRepositoryProviderInterface';
+
+@provider({
+  identifier: DistributionRepositoryInterfaceResolver,
+})
+export class DistributionRepositoryProvider implements DistributionRepositoryInterface {
+  private readonly table = 'observatory.monthly_distribution';
+  private readonly insert_procedure = 'observatory.insert_monthly_distribution';
+  private readonly today = new Date();
```

Comment:
**remarque**
Pas sur que les variables de class soient réévaluées à chaque requête mais je pense que c'est pas important dans ton cas

Reply:
Je suis d'accord avec Julien et je pense que comme tu utilise `this.endTime` qui dépend lui même de `this.today` il me semble que ça va poser des soucis. Si tu veux garder la même implémentation, tu peux utiliser un getter, par exemple : 
```typescript=
get today() {
  return new Date();
}
```
Par ailleurs, tu peux utiliser `date-fns` pour gérer les soustraction de date (pas obligatoire mais bon comme elle est déjà utilisée...)